internal/host/host.go

```diff
@@ -534,6 +535,9 @@ func (m *Manager) UpdateImageStatus(ctx context.Context, h *models.Host, newImag
 		}
 
 		m.log.Infof("Updating image status for %s with status %s to host %s", newImageStatus.Name, newImageStatus.Result, h.ID.String())
+		newImageStatus.DownloadRate = math.Max(hostImageStatuses[newImageStatus.Name].DownloadRate, newImageStatus.DownloadRate)
```

Comment:
maybe we should add those values also to the log

Reply:
I am not sure it is necessary. It could just confuse.

Reply:
This seems irrelevant to the actual fix (in validator.go).

Reply:
This will always persist the max doenloadRate.
So, if once the download rate was high it will never be updated to a lower value (in case the network was fixed...)

Reply:
We said that we will store the download rate only once.
If it exists in the db, we should not update